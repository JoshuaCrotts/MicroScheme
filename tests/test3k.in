
// Testing short circuiting

int counter;

int f(int x, int y) {
    counter++;
    return x+y;
}

void main() {
    int a = 13;
    int b = 11;
    int c = 18;
    int d = 17;
    int e = 23;

    counter=0;
    
    if ((a == f(b,2)) || (f(c,-1) == d)) prints("Yes\n");
    else prints("No\n");
    printd(counter);
    prints("\n");
    
    if ((a != f(b,2)) || (f(c,-1) == d)) prints("Yes\n");
    else prints("No\n");
    printd(counter);
    prints("\n");

    if ((a == f(b,2)) || (f(c,-1) != d)) prints("Yes\n");
    else prints("No\n");
    printd(counter);
    prints("\n");

    if ((a != f(b,2)) || (f(c,-1) != d)) prints("Yes\n");
    else prints("No\n");
    printd(counter);
    prints("\n");
}
