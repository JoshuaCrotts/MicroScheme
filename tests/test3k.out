Yes
1
Yes
3
Yes
4
No
6
